api/src/scripts/joiGenerator.ts

```diff
@@ -0,0 +1,148 @@
+  let variablePosition = text.search(regex);
+
+  if (variablePosition === -1) {
+    const functionRegex = new RegExp(`function\\s+${variableName}\\s?(`, "i");
+    variablePosition = text.search(functionRegex);
+    isFunction = true;
+  }
+  const fnStart = text.slice(variablePosition);
+
+  try {
+    [...fnStart].forEach((c) => {
+      switch (c) {
+        case "{":
+          curlyCount += 1;
+          curlyTotalCount += 1;
+          start = true;
+          break;
+        case "}":
+          curlyCount -= 1;
+          break;
+        case "(":
+          bracketCount += 1;
+          bracketTotalCount += 1;
+          start = true;
+          break;
+        case ")":
+          bracketCount -= 1;
+          break;
+        default:
+      }
+
+      if (breakCycle === true && c.search(/\S/) !== -1) {
+        breakCycle = false;
+      }
+
+      if (breakCycle !== true || c === ";") {
+        extractedText += c;
+      }
+
+      if (breakCycle === true) {
+        // eslint-disable-next-line @typescript-eslint/no-throw-literal
+        throw BreakException;
+      }
+
+      if (
+        start &&
+        curlyCount === 0 &&
+        bracketCount === 0 &&
+        !(isFunction && curlyTotalCount === 0 && bracketTotalCount === 1)
+      ) {
+        breakCycle = true;
+      }
+    });
+  } catch (e) {
+    if (e !== BreakException) throw e;
+  }
+
+  return extractedText;
+};
+
+const generate = () => {
+  const args = process.argv.slice(2);
+
+  let fileName = "";
+  let replace = false;
+  if (args) {
+    args.forEach((arg) => {
+      if (arg === "r" || arg === "R" || arg === "rewrite") {
+        replace = true;
+      } else {
+        fileName = arg;
+      }
+    });
+  }
+
+  if (fileName.length === 0) {
+    throw Error("filename not specified. Run e.g.: npm run generate project_create.ts");
+  }
+
+  const filePath = `./src/${fileName}`;
+  const fileContent = fs.readFileSync(filePath, "utf8");
+
+  // eslint-disable-next-line global-require
+  const { mkSwaggerSchema } = require(`../${fileName.replace(".ts", "")}`);
+
+  if (!mkSwaggerSchema) {
+    console.error(`File ${fileName} doesn't have exported function mkSwaggerSchema.
+    Try adding "export function mkSwaggerSchema" to the file.`);
+
+    return;
+  }
+
+  const swagger = mkSwaggerSchema(fastifyMock as any);
+  const joi = parseSwagger(swagger.schema, undefined, "2.0", { singleQuote: false, overrideKeys });
+
+  const found = extractVariable(fileContent, "requestBodyV1Schema");
+
+  const diff = Diff.diffChars(`${found}`, `const requestBodyV1Schema = ${joi.body};`);
+
+  diff.forEach((part) => {
+    // green for additions, red for deletions
+    // grey for common parts
+    // eslint-disable-next-line no-nested-ternary
+    const color = part.added ? "green" : part.removed ? "red" : "grey";
+    const rightColor = colors[color];
+    process.stderr.write(rightColor(part.value));
+  });
+
+  console.log();
```

Comment:
please remove the console.log()

Reply:
Actually there is reason for this console.log, to end the writing stream of text into console. This approach is suggested in examples of the library https://www.npmjs.com/package/diff#examples

Reply:
ups okay, nevermind then